### PR TITLE
Improve partition diff mode

### PR DESCRIPTION
**Pull request by zzeekk**

### CODE REVIEW

File: src/main/scala/io/smartdatalake/util/hdfs/Partition.scala [hunk 1]
```diff
@@ -54,6 +54,21 @@ private[smartdatalake] case class PartitionValues(elements: Map[String, Any]) {
 private[smartdatalake] object PartitionValues {
   val singleColFormat = "<partitionColName>=<partitionValue>[,<partitionValue>,...]"
   val multiColFormat = "<partitionColName1>=<partitionValue>,<partitionColName2>=<partitionValue>[;(<partitionColName1>=<partitionValue>,<partitionColName2>=<partitionValue>;...]"
+  def getOrdering(partitions: Seq[String]): Ordering[PartitionValues] = new Ordering[PartitionValues] {
```
Comment (kaelte) [inline src/main/scala/io/smartdatalake/util/hdfs/Partition.scala]: What is the purpose of this method?
What is meant by partitions and PartitionValues?
I was not able to find an explanation in scala doc.

Comment (zzeekk) [reply src/main/scala/io/smartdatalake/util/hdfs/Partition.scala]: @kaelte: I added scaladoc and tests for it. Can you please check again?

File: src/main/scala/io/smartdatalake/util/hdfs/Partition.scala [hunk 2]
```diff
@@ -54,6 +54,14 @@ private[smartdatalake] case class PartitionValues(elements: Map[String, Any]) {
...
+  /**
+   * Defines an Ordering for sorting PartitionValues.
+   * Sorting a list of partition values is only possible, if the partition columns to be considered are defined.
+   * As PartitionValues is a generic structure, the type of a value needs to be inferred for comparision.
+   * @param partitions partition columns to use for sorting
+   * @return Ordering to be used e.g. with Seq.sort|sortBy
+   */
   def getOrdering(partitions: Seq[String]): Ordering[PartitionValues] = new Ordering[PartitionValues] {
     def compare(pv1: PartitionValues, pv2: PartitionValues): Int = {
       partitions.toStream.map{
```
Comment (kaelte) [inline src/main/scala/io/smartdatalake/util/hdfs/Partition.scala]: Why converting to stream ?
```suggestion
      partitions.map{
```
Would a sequence not be lazy enough?

File: src/test/scala/io/smartdatalake/util/hdfs/PartitionValuesTest.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/*
```
Comment (kaelte) [inline src/test/scala/io/smartdatalake/util/hdfs/PartitionValuesTest.scala]: I will add some test case.
@zzeekk  please have a look!

**Comment by kaelte** [comment]
@zzeekk, please add commit https://github.com/smart-data-lake/smart-data-lake/commit/a22fcfd1159a3532eeb9853843f74b3e987adef2 to this pull request.

**Comment by zzeekk** [comment]
@kaelte: Thanks, additional tests look good.

I need to switch this PR to another #22 because i had to rewrite some commits because of wrong email.

**Pull request closed without merging**